sound/soc/soc-dai.c

```diff
@@ -617,8 +617,14 @@ int snd_soc_pcm_dai_prepare(struct snd_pcm_substream *substream)
 static int soc_dai_trigger(struct snd_soc_dai *dai,
 			   struct snd_pcm_substream *substream, int cmd)
 {
+	struct snd_soc_pcm_stream *stream;
 	int ret = 0;
 
+	stream = snd_soc_dai_get_pcm_stream(dai, substream->stream);
+	/* channels_max = 0 means the stream is not supported, return directly */
+	if (!stream->channels_max)
+		return 0;
+
```

Comment:
I would use this instead @bardliao :

````c
/*
 * snd_soc_dai_stream_valid() - check if a DAI supports the given stream
 *
 * Returns true if the DAI supports the indicated stream type.
 */
bool snd_soc_dai_stream_valid(struct snd_soc_dai *dai, int dir)
{
	struct snd_soc_pcm_stream *stream = snd_soc_dai_get_pcm_stream(dai, dir);

	/* If the codec specifies any channels at all, it supports the stream */
	return stream->channels_min;
}
````

Reply:
> I would use this instead @bardliao :
> 
> ```c
> /*
>  * snd_soc_dai_stream_valid() - check if a DAI supports the given stream
>  *
>  * Returns true if the DAI supports the indicated stream type.
>  */
> bool snd_soc_dai_stream_valid(struct snd_soc_dai *dai, int dir)
> {
> 	struct snd_soc_pcm_stream *stream = snd_soc_dai_get_pcm_stream(dai, dir);
> 
> 	/* If the codec specifies any channels at all, it supports the stream */
> 	return stream->channels_min;
> }
> ```

Yes, it is much better.